### PR TITLE
Increase config-generator unit test coverage from 20% to 46%

### PR DESCRIPTION
**Pull request by joshua-bone**
/lint
/ok-to-test

**What this PR does, why we need it**:
Adds unit tests for nearly all functions in main.go.

### CODE REVIEW

File: tools/config-generator/main_test.go [hunk 1]
```diff
@@ -43,3 +54,869 @@ func TestOutputConfig(t *testing.T) {
...
+		yaml.MapItem{Key: "report_template", Value: "Report Template"},
+		yaml.MapItem{Key: "foo", Value: []interface{}{"bar", "baz"}},
+	}
+	reporter_config := yaml.MapSlice{
+		yaml.MapItem{Key: "slack", Value: slack},
+	}
+
+	repo_name := "foo_repo"
+	repositories = []repositoryData{
+		repositoryData{Name: repo_name, EnablePerformanceTests: false},
```
Comment (mattmoor) [inline tools/config-generator/main_test.go:400]: Format Go code:
```suggestion
		{Name: repo_name, EnablePerformanceTests: false},
```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

File: tools/config-generator/main_test.go [hunk 2]
```diff
@@ -43,3 +54,869 @@ func TestOutputConfig(t *testing.T) {
...
+		yaml.MapItem{Key: "report_template", Value: "Report Template"},
+		yaml.MapItem{Key: "foo", Value: []interface{}{"bar", "baz"}},
+	}
+	reporterConfig := yaml.MapSlice{
+		yaml.MapItem{Key: "slack", Value: slack},
+	}
+
+	repoName := "foo_repo"
+	repositories = []repositoryData{
+		repositoryData{Name: repo_name, EnablePerformanceTests: false},
```
Comment (mattmoor) [inline tools/config-generator/main_test.go:400]: Format Go code:
```suggestion
		{Name: repo_name, EnablePerformanceTests: false},
```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

File: tools/config-generator/main_test.go (hunk 2 above)
Comment (mattmoor) [inline tools/config-generator/main_test.go:400]: Format Go code:
```suggestion
		{Name: repo_name, EnablePerformanceTests: false},
```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

**Comment by chaodaiG** [comment]
@googlebot I consent

**Comment by joshua-bone** [comment]
@googlebot I consent

**Comment by joshua-bone** [comment]
/close
Switching to new branch for cleaner commit history.

**Pull request closed without merging**